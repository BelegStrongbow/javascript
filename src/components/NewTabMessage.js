import React from "react";
import { FormattedMessage } from "react-intl";
import styled from "styled-components";

const NewTabMessageContainer = styled.span`
	// Safari+VoiceOver bug see issue 445 and 712.
	padding-top: 0.5em;
`;

/**
 * Returns the rendered Opens in a New Tab Message component.
 *
 * Note about recommended usage: use it as a value to pass to a react-intl
 * FormattedMessage placeholder or in any other way that the final result will
 * make this text inline with the text is added to.
 *
 * @returns {ReactElement} The rendered NewTabMessage component.
 */
export default function NewTabMessage() {
	return (
		<NewTabMessageContainer className="screen-reader-text">
			<FormattedMessage id="link.new-window" defaultMessage="(Opens in a new browser tab)" />
		</NewTabMessageContainer>
	);
}
