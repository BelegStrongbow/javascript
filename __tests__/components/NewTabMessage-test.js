import React from 'react';
import NewTabMessage from '../../src/components/NewTabMessage';
import { createComponentWithIntl } from "../../utils";

test('the new tab message component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<NewTabMessage />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
