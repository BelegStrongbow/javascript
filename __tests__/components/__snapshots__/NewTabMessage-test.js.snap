exports[`test the new tab message component matches the snapshot 1`] = `
<span
  className="screen-reader-text fuqaAG">
  <span>
    (Opens in a new browser tab)
  </span>
</span>
`;
